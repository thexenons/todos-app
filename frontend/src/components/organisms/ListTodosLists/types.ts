import { TodosList } from "../../../types";

export interface ListTodosListsProps {
	todosLists: TodosList[];
	onClickTodosList?: (todosList: TodosList) => void;
	onBack?: () => void;
}
