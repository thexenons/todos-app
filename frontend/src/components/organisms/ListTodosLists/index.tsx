import { FC } from "react";

import { ListTodosListsProps } from "./types";

const ListTodosLists: FC<ListTodosListsProps> = ({
	todosLists,
	onClickTodosList,
	onBack,
}) => (
	<div>
		<h2>Todos List {onBack && <button onClick={onBack}>Back</button>}</h2>
		<ul>
			{todosLists.map((todosList) => (
				<li
					key={todosList.id}
					role="presentation"
					onClick={() => onClickTodosList?.(todosList)}
				>
					<h3>{todosList.name}</h3>
				</li>
			))}
		</ul>
	</div>
);

export default ListTodosLists;
