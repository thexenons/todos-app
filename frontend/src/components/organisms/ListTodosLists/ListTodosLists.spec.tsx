import { render } from "@testing-library/react";

import BaseTestComponent from "../../../tests/BaseTestComponent";
import ListTodosLists from ".";

describe("ListTodosLists", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<ListTodosLists todosLists={[]} />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
