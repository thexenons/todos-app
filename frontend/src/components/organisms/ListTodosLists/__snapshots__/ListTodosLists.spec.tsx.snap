// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListTodosLists > Render 1`] = `
<div>
  <div>
    <h2>
      Todos List 
    </h2>
    <ul />
  </div>
</div>
`;
